src/awkward/_v2/contents/content.py

```diff
@@ -924,6 +924,51 @@ def validityerror(self, path="layout"):
     def purelist_parameter(self, key):
         return self.Form.purelist_parameter(self, key)
 
+    def is_unique(self, axis=None):
+        negaxis = axis if axis is None else -axis
+        starts = ak._v2.index.Index64.zeros(1, self.nplike)
+        parents = ak._v2.index.Index64.zeros(len(self), self.nplike)
+        return self._is_unique(negaxis, starts, parents, 1)
+
+    def unique(self, axis=None):
+        if axis == -1 or axis is None:
+            negaxis = axis if axis is None else -axis
+            if negaxis is not None:
+                branch, depth = self.branch_depth
+                if branch:
+                    if negaxis <= 0:
+                        raise ValueError(
+                            "cannot use non-negative axis on a nested list structure "
+                            "of variable depth (negative axis counts from the leaves "
+                            "of the tree; non-negative from the root)"
+                        )
+                    if negaxis > depth:
+                        raise ValueError(
+                            "cannot use axis={0} on a nested list structure that splits into "
+                            "different depths, the minimum of which is depth={1} from the leaves".format(
+                                axis, depth
+                            )
+                        )
+                else:
+                    if negaxis <= 0:
+                        negaxis = negaxis + depth
+                    if not (0 < negaxis and negaxis <= depth):
+                        raise ValueError(
+                            "axis={0} exceeds the depth of the nested list structure "
+                            "(which is {1})".format(axis, depth)
+                        )
```

Comment:
All errors in which a user specifies an unusable axis (e.g. the three above) should use `np.AxisError` now—it's a convention I'm trying to establish, and in overlaps with NumPy, it's what NumPy does.

The "axis too deep" error should have this text:

```python
np.AxisError("axis={0} exceeds the depth of this array ({1})".format(axis, depth))
```

Cross-linking the comment to @stormiestsin: https://github.com/scikit-hep/awkward-1.0/pull/1135#discussion_r744811036

Reply:
done